### PR TITLE
Application Rejection Handling

### PR DESCRIPTION
**Pull request by NClin**
Following https://github.com/LemmyNet/lemmy-ui/issues/1096

It appears approve.rs wasn't handling rejected applications. This change adds a default Application Denied email, and deletes the account in question.

Change is not verified due to environment issues - seeking help on this to get them validated (I'm so close! Running locally, but the login button doesn't seem to do anything). Also new to pull requests so any advice appreciated.

Also requires strings to be added to the translations submodule.

### CODE REVIEW

File: crates/api/src/site/registration_applications/approve.rs [hunk 1]
```diff
@@ -44,15 +44,23 @@ impl Perform for ApproveRegistrationApplication {
...
         send_application_approved_email(&approved_local_user_view, context.settings())?;
       }
+    } else {
+      let denied_local_user_view = LocalUserView::read(context.pool(), assessed_user_id).await?;
+    
+      if denied_local_user_view.local_user.email.is_some() {
+        send_application_denied_email(&denied_local_user_view, context.settings())?;
+      }
+      
+      LocalUser::delete(context.pool(), assessed_user_id).await?;
```
Comment (Nutomic) [inline crates/api/src/site/registration_applications/approve.rs:63]: You also have to delete from Person table.

File: crates/api_common/src/utils.rs [hunk 2]
```diff
@@ -472,6 +472,20 @@ pub fn send_application_approved_email(
   send_email(&subject, email, &user.person.name, &body, settings)
 }
 
+
+/// IMPLEMENT ME
```
Comment (Nutomic) [inline crates/api_common/src/utils.rs:476]: Old comment?

File: crates/api_common/src/utils.rs [hunk 3]
```diff
@@ -472,6 +472,20 @@ pub fn send_application_approved_email(
...
+pub fn send_application_denied_email(
+  user: &LocalUserView,
+  settings: &Settings,
+) -> Result<(), LemmyError> {
+  let email = &user.local_user.email.clone().expect("email");
```
Comment (Nutomic) [inline crates/api_common/src/utils.rs:481]: Dont use expect.

File: crates/api/src/site/registration_applications/approve.rs (hunk 1 above)
Comment (dessalines) [reply crates/api/src/site/registration_applications/approve.rs:63]: Not sure about this one. Deleting those from the tables does free up the IDs, but it also means we can't show them their deny reason, which we did before. I spose I'm good with it, if they didn't provide an email.

File: crates/api_common/src/utils.rs (hunk 3 above)
Comment (dessalines) [reply crates/api_common/src/utils.rs:481]: You'll see this was copy-pasted from the approval send, which also unfortunately uses expect. 

It does check for the email before this is called, but I agree, using if let is safer.

File: crates/api/src/site/registration_applications/approve.rs (hunk 1 above)
Comment (dessalines) [reply crates/api/src/site/registration_applications/approve.rs:63]: Another thing about this: deleting their person will actually delete the entire registration application from the DB. Nothing will be known about their deny.

**Comment by dessalines** [comment]
Actually I'll close this as #3050 does the same thing.

**Pull request closed without merging**